be/src/exec/vectorized/topn_node.cpp

```diff
@@ -234,21 +256,22 @@ pipeline::OpFactories TopNNode::decompose_to_pipeline(pipeline::PipelineBuilderC
                                                                            SIZE_OF_CHUNK_FOR_FULL_SORT);
     }
 
-    // add sort operator to this pipeline
-    auto ope = std::make_shared<SortSinkOperatorFactory>(context->next_operator_id(), id(), chunks_sorter,
-                                                         _sort_exec_exprs, _order_by_types, _materialized_tuple_desc,
-                                                         child(0)->row_desc(), _row_descriptor);
-    operator_sink_with_sort.emplace_back(std::move(ope));
-    context->add_pipeline(operator_sink_with_sort);
+    // add SortSinkOperator to this pipeline
+    auto sort_sink_operator = std::make_shared<SortSinkOperatorFactory>(
+            context->next_operator_id(), id(), chunks_sorter, _sort_exec_exprs, _order_by_types,
+            _materialized_tuple_desc, child(0)->row_desc(), _row_descriptor);
+    pred_operators.emplace_back(std::move(sort_sink_operator));
+    context->add_pipeline(pred_operators);
```

Comment:
pred_operators have been added into context twice

Reply:
Only when source_operator->num_driver_instances() > 1, a LocalExchangeSinkOperator is added to pred_operators, so current pipeline comes to end, then pred_operators is used for a new pipeline: LocalExchangeSourceOperator->SortSinkOperator.